### PR TITLE
Prepopulate audit form field with audited url.

### PR DESCRIPTION
**Pull request by ckundo**
- use the params hash to render the url query param
- add unit tests and refactor report
- use a shared partial for the audit form in index and results templates

### CODE REVIEW

File: app/views/shared/_audit_form.html.erb [hunk 1]
```diff
@@ -1,18 +1,9 @@
...
-    </form>
-    <div>
-      <% if signed_out? %>
-        <%= link_to "Sign up for a free trial of AccessLint Pro", sign_up_path %>
-      <% end %>
-    </div>
+<form class="auditform" action="/results" method="get" novalidate>
+  <label for="url">URL to Test</label>
+  <div class="auditform-formgroup">
+    <%= url_field_tag :url, URI(params[:url].to_s), placeholder: 'http://www.example.com',
```
Comment (ckundo) [inline app/views/shared/_audit_form.html.erb]: This is where the magic happens! We're wondering if `params[:uri]` is vulnerable to script injection here (that's why we used the `URI` module to parse the param).

Comment (mike-burns) [inline app/views/shared/_audit_form.html.erb]: It is passed as an attribute to the `#tag` method as an attribute value, which is passed to `#tag_options` with `escape` set to true. This causes it to escape the value.

If you want to make sure, you could add a test with an injection. Here's my attempt:

`?url=-- "> --><script>alert('yo')</script>`

**Comment by mike-burns** [comment]
I can't think of how to hack that. LGTM.

File: app/views/shared/_audit_form.html.erb (hunk 1 above)
Comment (georgebrock) [inline app/views/shared/_audit_form.html.erb]: I just read the same bits of the Rails source, and I agree with @mike-burns: this should be safe.

(If you *do* find a way to inject a script, it'd probably mean you'd discovered a pretty major Rails security bug)

**Pull request merged**